Update troubleshooting guides with debug service usage (#41396)
gravitational/teleport: docs/pages/management/diagnostics/profiles.mdx

profiling data. Profiles can help identify the cause of spikes in CPU,
the source of memory leaks, or the reason for a deadlock.

## Using `teleport debug profile`

(!docs/pages/includes/diagnostics/debug-service-intro.mdx!)

`teleport debug profile` collects a list of pprof profiles. It outputs a
compressed tarball (`.tar.gz`) to STDOUT. You decompress it using `tar` or
direct the result to a file.

By default, it collects `goroutine`, `heap` and `profile` profiles.

```bash
# Collect default profiles and save to a file.
$ teleport debug profile > pprof.tar.gz
$ tar xvf pprof.tar.gz

# Collect default profiles and decompress it.
$ teleport debug profile | tar xzv -C ./

# Collect "trace" and "mutex" profiles and save to a file.
$ teleport debug profile trace,mutex > pprof.tar.gz

> ```

Reviewer: Could we add an example (or reuse one of the examples above) with `seconds`?

nit: I also really like examples below where we show the `go tool pprof` command to visualize the result and i think it would be great if we can add them here too.

Reviewer: We should link the CLI reference for `teleport debug` somewhere in this page.

Author: I've added examples below this bash block.

profiling data. Profiles can help identify the cause of spikes in CPU,
the source of memory leaks, or the reason for a deadlock.

## Using `teleport debug profile`

(!docs/pages/includes/diagnostics/debug-service-intro.mdx!)

`teleport debug profile` collects a list of pprof profiles. It outputs a
compressed tarball (`.tar.gz`) to STDOUT. You decompress it using `tar` or
direct the result to a file.

By default, it collects `goroutine`, `heap` and `profile` profiles.

Each profile collected will have a correspondent file inside the tarball. For
example, collecting `goroutine,trace,heap` will result in `goroutine.pprof`,
`trace.pprof`, and `heap.pprof` files.

> ```bash

Reviewer 2: ```suggestion
```code
```
We have some custom logic enabled by the `code` label

# Collect default profiles and save to a file.
$ teleport debug profile > pprof.tar.gz
$ tar xvf pprof.tar.gz

# Collect default profiles and decompress it.
$ teleport debug profile | tar xzv -C ./

# Collect "trace" and "mutex" profiles and save to a file.
$ teleport debug profile trace,mutex > pprof.tar.gz

# Collect profiles setting the profiling time in seconds
$ teleport debug profile -s 20 trace > pprof.tar.gz
```

If you're running Teleport on a Kubernetes cluster you can directly collect
profiles to a local directory without an interactive session:

```bash
$ kubectl -n teleport exec my-pod -- teleport debug profile > pprof.tar.gz 
```

After extracting the contents, you can use `go tool` commands to explore and
visualize them:

> ```bash

Reviewer 2: ```suggestion
```code
```

profiling data. Profiles can help identify the cause of spikes in CPU,
the source of memory leaks, or the reason for a deadlock.

> ## Using Debug Service

Reviewer 2: ```suggestion
## Using the Debug Service
```

> Teleport's Debug Service enables administrators to collect diagnostic profiles

Reviewer 2: ```suggestion
The Teleport Debug Service enables administrators to collect diagnostic profiles
```

Each profile collected will have a correspondent file inside the tarball. For
example, collecting `goroutine,trace,heap` will result in `goroutine.pprof`,
`trace.pprof`, and `heap.pprof` files.

```code
# Collect default profiles and save to a file.
$ teleport debug profile > pprof.tar.gz
$ tar xvf pprof.tar.gz

# Collect default profiles and decompress it.
$ teleport debug profile | tar xzv -C ./

# Collect "trace" and "mutex" profiles and save to a file.
$ teleport debug profile trace,mutex > pprof.tar.gz

# Collect profiles setting the profiling time in seconds
$ teleport debug profile -s 20 trace > pprof.tar.gz
```

If you're running Teleport on a Kubernetes cluster you can directly collect
profiles to a local directory without an interactive session:

```code
$ kubectl -n teleport exec my-pod -- teleport debug profile > pprof.tar.gz 
```

After extracting the contents, you can use `go tool` commands to explore and
visualize them:

```code
# Opens the terminal interactive explorer
$ go tool pprof heap.pprof

# Opens the web visualizer
$ go tool pprof -http : heap.pprof

# Visualize trace profiles
$ go tool trace trace.pprof
```

> (!docs/pages/includes/diagnostics/teleport-debug-config.mdx!)

Reviewer: `teleport-debug-config.mdx` can be placed up a little? maybe before kube instruction or before visualization section?
